app/src/main/java/com/lagradost/quicknovel/ui/result/ResultFragment.kt

```diff
@@ -727,6 +870,52 @@ class ResultFragment : Fragment() {
             }
         }
 
+        // Update chapter list based on filter state
+        fun updateChapterList() {
```

Comment:
Why is this done inside the ResultFragment? Instead of calling viewModel.getFilteredChapters() it should properly set the data inside the viewmodel for the Fragment to observe, not the other way around.